### PR TITLE
multitenant registry

### PR DESCRIPTION
**Pull request by famartinrh**
Basic deployment of multitenant registry:
 - multitenancy enabled
 - tenant-manager deployment
 - basic postgresql deployment
 - simple ingresses useful for local testing

### CODE REVIEW

**Comment by carlesarnal** [comment]
LGTM, just one curiosity, do we want to use `latest-snapshot` as the base image? (I assume yes, just curious)

**Comment by famartinrh** [comment]
I assumed we want to use the latest version to get all the upcoming changes on limits and auth...

**Comment by famartinrh** [comment]
I added a initialization script to the db to create multiple databases in the same postgresql deployment

**Comment by EricWittmann** [comment]
> 
> 
> LGTM, just one curiosity, do we want to use `latest-snapshot` as the base image? (I assume yes, just curious)

Yes for sure.  At least for now.  We will need to think about some orthogonal CI/CD approach if we ever have `master` diverge from what we want in MT/MAS.  But for now I think `latest-snapshot` is appropriate.

**Comment by EricWittmann** [comment]
This looks great to me.  @paoloantinori any issue with this?

File: README.md [hunk 1]
```diff
@@ -16,3 +16,19 @@ To apply the `kustomization` command locally, to verify what is the output of th
 cd manifests/registry
 oc get -k ./overlay_limitador/ -o yaml | tee log.txt
 ```
+
+## Multitenancy enabled deployment
+
+The `overlay_multitenancy` contains configuration files for deploying Apicurio Registry with multitenancy enabled.
+
+```bash
+./kustomize build ./manifests/registry/overlay_multitenancy/ | k apply -f -
```
Comment (paoloantinori) [inline README.md:25]: what's `k`?

**Review by paoloantinori — COMMENTED** [review]
I might be misreading this, but I see that there is a ` /opt/app-root/src/postgresql-start/db-init-script-configmap` that gets invoked automatically by some supported mechanism in the `centos7/postgresql-12-centos7:1` .

I see that you use a PersistenVolume and that's great.

But...

What is not explicit from that interaction is if that script is invoked only no db is found or not. I don't see any safety logic in the script, so it might be in the base image, but I'm not sure.

**Comment by paoloantinori** [comment]
Addtionally:
- you are using ConfigMap for all the user/pass besides the DB admin one, maybe it's better to externalize those too.

And, not required now, but for the future, we should consider looking in the supported way to have the password stored encrypted  directly on the repo, but I might need to share with you  the key I've used so far.

It can probably be deployed as is already and refined in the future, the only thing that I'd really like to get a comment on is if the init script is automatically triggered at every pod redeployment, or if it skips the step if it finds the db already populated (since we are storing it on a PersistentVolume)

Other than that, great work!

File: README.md (hunk 1 above)
Comment (paoloantinori) [reply README.md:25]: the command could be changed to `oc apply -k ./manifests/registry/overlay_multitenancy/`

**Comment by paoloantinori** [comment]
I've tried to deploy this directly on Operate First:

the UI of registry stops rendering.


Additionally:
I've tried to kill the database pod to check what is the supposed behavior in relation to the volume, and it fails badle, going in CrashLoop:

```
waiting for server to start....2021-05-22 19:19:54.694 UTC [25] LOG:  starting PostgreSQL 12.5 on x86_64-redhat-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44), 64-bit
2021-05-22 19:19:54.694 UTC [25] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2021-05-22 19:19:54.696 UTC [25] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2021-05-22 19:19:54.729 UTC [25] LOG:  redirecting log output to logging collector process
2021-05-22 19:19:54.729 UTC [25] HINT:  Future log output will appear in directory "log".
. done
server started
/var/run/postgresql:5432 - accepting connections
=> sourcing /opt/app-root/src/postgresql-start/create-databases.sh ...
Multiple database creation requested: apicurio,tenantmanager,fleetmanager
  Creating user and database 'apicurio'
ERROR:  role "apicurio" already exists
```

File: README.md (hunk 1 above)
Comment (famartinrh) [reply README.md:25]: my bad, `k` is my alias for `kubectl`

**Comment by famartinrh** [comment]
great feedback @paoloantinori , I did no check if that script was safe for database restarts, I'll improve it...
about the database passwords, I completely agree all of that should be on secrets and encrypted, or externalized as you said, but I don't know what you mean exactly by externalized.

**Comment by paoloantinori** [comment]
> about the database passwords, I completely agree all of that should be on secrets and encrypted, or externalized as you said, but I don't know what you mean exactly by externalized.

just meant encapsulated in their own Secret or ConfigMap

**Pull request merged**